vegbilder-client/src/components/MapContainer/MapContainer.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Map, TileLayer } from 'react-leaflet';
+import 'leaflet/dist/leaflet.css';
+import { useRecoilState, useRecoilValue } from 'recoil';
+import { useLeafletBounds } from 'use-leaflet';
+
+import { crsUtm33N } from './crs';
+import ImagePointLayersWrapper from 'components/ImagePointsLayersWrapper/ImagePointsLayersWrapper';
+import MapControls from 'components/MapControls/MapControls';
+import { useCurrentCoordinates } from 'contexts/CurrentCoordinatesContext';
+import { useLoadedImagePoints } from 'contexts/LoadedImagePointsContext';
+import { currentYearState } from 'recoil/atoms';
+import getImagePointsInTilesOverlappingBbox from 'apis/VegbilderOGC/getImagePointsInTilesOverlappingBbox';
+import { IImagePoint, ILatlng } from 'types';
+import { findNearestImagePoint } from 'utilities/imagePointUtilities';
+import { createSquareBboxAroundPoint, isBboxWithinContainingBbox } from 'utilities/latlngUtilities';
+import { availableYearsQuery } from 'recoil/selectors';
+import { useCurrentImagePoint } from 'contexts/CurrentImagePointContext';
+import { settings } from 'constants/constants';
+import useQueryParamState from 'hooks/useQueryParamState';
+
+interface IMapContainerProps {
+  showMessage: (message: string) => void;
+}
+
+const MapContainer = ({ showMessage }: IMapContainerProps) => {
+  const { currentCoordinates, setCurrentCoordinates } = useCurrentCoordinates();
+  const [[south, west], [north, east]] = useLeafletBounds();
+  const [isFetching, setIsFetching] = useState(false);
+  const { loadedImagePoints, setLoadedImagePoints } = useLoadedImagePoints();
+  const [currentYear, setCurrentYear] = useRecoilState(currentYearState);
+  const availableYears = useRecoilValue(availableYearsQuery);
+  const { setCurrentImagePoint } = useCurrentImagePoint();
+  const [, setQueryParamYear] = useQueryParamState('year');
+
+  const createBboxForVisibleMapArea = useCallback(() => {
+    // Add some padding to the bbox because the meridians do not perfectly align with the vertical edge of the screen (projection issues)
+    let paddingX = (east - west) * 0.1;
+    let paddingY = (north - south) * 0.1;
+    if (settings.debugMode) {
+      // Simulate a smaller visible map area for debugging purposes.
+      paddingX = (west - east) * 0.2;
+      paddingY = (south - north) * 0.2;
+    }
+
+    return {
+      south: south - paddingY,
+      west: west - paddingX,
+      north: north + paddingY,
+      east: east + paddingX,
+    };
+  }, [south, west, north, east]);
+
+  const fetchImagePointsFromNewestYearWhereUserClicked = (latlng: ILatlng) => {
```

Comment:
Kan ikke `fetchImagePointsFromNewestYearWhereUserClicked` bli async selv i stedet for å opprette en async funksjon inni og kalle den umiddelbart?

Reply:
Får klage fra editoren da :/